client/src/main/java/org/apache/rocketmq/client/impl/consumer/ConsumeMessageOrderlyService.java

```diff
@@ -92,10 +92,22 @@ public void run() {
         }
     }
 
-    public void shutdown() {
+    @Override
+    public void shutdown(long awaitTerminateMillis) {
         this.stopped = true;
         this.scheduledExecutorService.shutdown();
         this.consumeExecutor.shutdown();
+        //await to consume
+        if (awaitTerminateMillis > 0) {
+            try {
+                this.consumeExecutor.awaitTermination(awaitTerminateMillis,TimeUnit.MILLISECONDS);
+                if (!this.consumeExecutor.isTerminated()) log.info("There are messages still being consumed in thread pool, but not going to await them anymore. Have awaited for {} ms",awaitTerminateMillis);
```

Comment:
Hi @Jaskey ,

I added a ThreadUtils, please refer to [here](https://github.com/apache/incubator-rocketmq/commit/e3f4251c91a73f4e51732bcb1690554ac5fb3096).